Gems/ROS2/Code/Source/Sensor/SensorHelpers.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ROS2/Sensor/Events/PhysicsBasedSource.h>
+#include <ROS2/Sensor/Events/TickBasedSource.h>
+#include <ROS2/Sensor/ROS2SensorComponentBase.h>
+#include <ROS2/Sensor/SensorHelper.h>
+
+namespace ROS2
+{
+
+    AZStd::vector<AZ::EntityComponentIdPair> GetSensorsForEntity(const AZ::EntityId& entityId)
+    {
+        AZStd::vector<AZ::EntityComponentIdPair> sensors;
+        AZ::Entity* entity = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->FindEntity(entityId);
+        AZ_Assert(entity, "Entity not found");
+        if (entity)
```

Comment:
```suggestion
        if (AZ::Entity* entity = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->FindEntity(entityId))
```
I think in case of invalid `entityId` we shouldn't both assert and check, only check and treat it as valid input

Reply:
I think both should stay.
- Assert causes breakpoint during debug
- if - prevents crash in the product.
Using only assert is not a great idea IHMO.

Reply:
I think the best solution would be both `AZ_Warning` and `if` for user can set breakpoint for debug. If we put `AZ_Assert` here, we can't skip it when debugging something else.

Reply:
Ok. Agree